db/migrate/20151014144004_host_collection_to_hosts.rb

```diff
@@ -0,0 +1,89 @@
+class HostCollectionToHosts < ActiveRecord::Migration
+  class Katello::System < ActiveRecord::Base
+    self.table_name = "katello_systems"
+  end
+
+  class Katello::HostCollection < ActiveRecord::Base
+    self.table_name = "katello_host_collections"
+  end
+
+  class Katello::SystemHostCollections < ActiveRecord::Base
+    self.table_name = "katello_system_host_collections"
+  end
+
+  class Katello::HostCollectionHosts < ActiveRecord::Base
+    self.table_name = "katello_host_collection_hosts"
+  end
```

Comment:
partha recently discovered that a lot of (i.e. all) of our migrations have been doing this incorrectly.  As a namespaced engine, we have to not use the actual Katello module, so they should look like:

```
  class HostCollection < ActiveRecord::Base
    self.table_name = "katello_host_collections"
  end
```

Also you'll need to define Host and define all the relationships between host, host collection, and system